std-bits/table/src/main/java/org/enso/table/data/column/builder/object/ObjectBuilder.java

```diff
@@ -99,11 +99,16 @@ public void setCurrentSize(int currentSize) {
   }
 
   private void grow() {
+    int desiredCapacity = 3;
     if (data.length > 1) {
-      grow(data.length * 3 / 2);
-    } else {
-      grow(3);
+      desiredCapacity = (data.length * 3 / 2);
+    }
+
+    if (currentSize >= desiredCapacity) {
+      desiredCapacity = currentSize + 1;
     }
+
+    grow(desiredCapacity);
```

Comment:
why bother resizing if `desiredCapacity` less than current size?

Reply:
Right? I would think the same. That's why the bug bit us :)

Normally we do:
```
if (currentSize >= capacity) { grow(); }
```

Where we reallocate to `3/2 * currentSize` so that the reallocations amortize nicely. Classic.

Problem is, this assumes `currentSize` does not grow too much, sometimes when adding nulls, `currentSize` can grwo much more and `3/2 * currentSize` will not be enough. That additional check is to ensure that if `currentSize` grew more than we expected, we can still fit the new element there.

I will add a comment.